manifests/install/solaris.pp

```diff
@@ -0,0 +1,53 @@
+# == Class puppet_agent::windows::install
+#
+# Private class called from puppet_agent class
+#
+# Manage the install process for windows specifically
+#
+class puppet_agent::install::solaris(
+  $package_version,
+  $install_options = [],
+){
+  assert_private()
+  if $::operatingsystemmajrelease == '10' {
+    $_unzipped_package_name = regsubst($::puppet_agent::prepare::package::package_file_name, '\.gz$', '')
+    $install_script = 'solaris_install.sh.erb'
+
+    # The following are expected to be available in the solaris_install.sh.erb template:
+    $adminfile = '/opt/puppetlabs/packages/solaris-noask'
+    $sourcefile = "/opt/puppetlabs/packages/${_unzipped_package_name}"
+    # Starting with puppet6 collections we no longer carry the mcollective service
+    if $::puppet_agent::collection != 'PC1' and $::puppet_agent::collection != 'puppet5' {
```

Comment:
invert the if for puppet6 an option?

Reply:
We could do that, but then we would need to add each new collection as they are released. This way should be future proof (i.e. no changes required for puppet7/8/9)

Reply:
That makes sense. Coolness.